File: Observer-Pattern/Weather-Station-App/CurrentConditionsDisplay.js
"use strict";
exports.__esModule = true;
var CurrentConditionsDisplay = /** @class */ (function () {
    function CurrentConditionsDisplay(weatherData) {
        this.weatherData = weatherData;
        this.weatherData.registerObserver(this);
    }
    CurrentConditionsDisplay.prototype.update = function (temperature, humidity, pressure) {
        this.temperature = temperature;
        this.humidity = humidity;
        this.pressure = pressure;
        this.display();
    };
    CurrentConditionsDisplay.prototype.display = function () {
        console.log("----CONDITIONS DISPLAY-----");
        console.log("Temperature is:", this.temperature);
        console.log("Humidity is:", this.humidity);
        console.log("Pressure is:", this.pressure);
    };
    return CurrentConditionsDisplay;
}());
exports.CurrentConditionsDisplay = CurrentConditionsDisplay;

File: Java/src/main/java/ObserverPattern/CurrentConditionsDisplay.java
package ObserverPattern;

public class CurrentConditionsDisplay extends WeatherStationDisplay{
    private int temperature;
    private int pressure;
    private int humidity;

    CurrentConditionsDisplay(){
        this.temperature = 0;
        this.pressure = 0;
        this.humidity = 0;
    }
    @Override
    public void update(int temperature, int pressure, int humidity) {
        this.temperature = temperature;
        this.pressure = pressure;
        this.humidity = humidity;
    }

    public void ShowCurrentConditions(){
        System.out.println("Temperature is : " + temperature);
        System.out.println("Pressure is : " + pressure);
        System.out.println("Humidity is : " + humidity);
    }
}
